Code/Tools/ProjectManager/Source/GemCatalog/GemInfo.cpp

```diff
@@ -100,4 +100,56 @@ namespace O3DE::ProjectManager
     {
         return (m_displayName < gemInfo.m_displayName);
     }
+
+    GemInfo::Platforms GemInfo::GetPlatformFromString(const QString& platformText)
+    {
+        if(platformText == "Windows")
+        {
+            return GemInfo::Platform::Windows;
+        }
+        else if(platformText == "Linux")
+        {
+            return GemInfo::Platform::Linux;
+        }
+        else if(platformText == "Android")
+        {
+            return GemInfo::Platform::Android;
+        }
+        else if(platformText == "iOS")
+        {
+            return GemInfo::Platform::iOS;
+        }
+        else if(platformText == "macOS")
+        {
+            return GemInfo::Platform::macOS;
+        }
+        else
+        {
+            return GemInfo::Platforms();
+        }
+    }
+
+    GemInfo::Platforms GemInfo::GetPlatformsFromStringList(const QStringList& platformStrings)
+    {
+        GemInfo::Platforms newPlatforms = GemInfo::Platforms();
+        for(const QString& platform : platformStrings)
+        {
+            newPlatforms |= GetPlatformFromString(platform);
+        }
+        return newPlatforms;
+    }
+
+    QStringList GemInfo::GetPlatformsAsStringList() const
+    {
+        QStringList platformStrings;
+        for (int i = 0; i < GemInfo::NumPlatforms; ++i)
+        {
+            const GemInfo::Platform platform = static_cast<GemInfo::Platform>(1 << i);
+            if(m_platforms & platform)
+            {
+                platformStrings << GetPlatformString(platform);
+            }
+        }
+        return platformStrings;
+    }
```

Comment:
I'm seeing this repeated code enough in the last couple of reviews I'm starting to think we might benefit from a utility `GetPlatformsAsStringList` function on the `GemInfo` class that takes in a bit mask and returns a QStringList instead of having this code copied in several places

Reply:
Although, now that I say that, most of those other places want the list in reverse order, which complicates things

Reply:
Yea, there's one instance actually where we want the list in reverse order, namely in `GemItemDelegate.cpp` in this PR: https://github.com/o3de/o3de/pull/13106

I think refactoring this method may require a survey of all instances where we do this kind of pattern. Without that data, it's hard to generalize this.